src/pinto_lab_to_nwb/behavior/interfaces/virmentemporalalignmentbehaviorinterface.py

```diff
@@ -0,0 +1,141 @@
+            The name of the sync data struct in the .mat file. (e.g. "wf_behav_sync_data")
+        im_frame_timestamps_name: Optional[str]
+            The name of the variable in the .mat file that contains the aligned timestamps for the imaging frames.
+            (e.g. "im_frame_timestamps")
+        """
+        self.verbose = verbose
+        super().__init__(file_path=file_path)
+        self.im_frame_timestamps_name = im_frame_timestamps_name or "im_frame_timestamps"
+        mat_dict = read_mat(filename=file_path)
+        assert sync_data_struct_name in mat_dict, f"'{sync_data_struct_name}' not in file '{file_path}'."
+        self._mat_dict = mat_dict[sync_data_struct_name]
+        self.sync_data_struct_name = sync_data_struct_name
+        self._times = None
+
+    def get_original_timestamps(self) -> np.ndarray:
+        mat_dict = read_mat(filename=self.source_data["file_path"])[self.sync_data_struct_name]
+        assert (
+            self.im_frame_timestamps_name in mat_dict
+        ), f"'{self.im_frame_timestamps_name}' not in in file '{self.source_data['file_path']}'."
+        return mat_dict[self.im_frame_timestamps_name]
+
+    def get_timestamps(self) -> np.ndarray:
+        return self._times if self._times is not None else self.get_original_timestamps()
+
+    def set_aligned_timestamps(self, aligned_timestamps: np.ndarray):
+        self._times = aligned_timestamps
+
+    def add_position(self, nwbfile: NWBFile):
+        # Processed position, velocity, viewAngle
+        position_by_im_frame_name = "position_by_im_frame"
+        if position_by_im_frame_name not in self._mat_dict:
+            return
+
+        behavior = get_module(nwbfile, "behavior", "contains processed behavioral data")
+        if "Position" not in behavior.data_interfaces:
+            behavior.add(Position(name="Position"))
+
+        position_obj = behavior.data_interfaces["Position"]
+
+        # reference "BehavSync2P.m" script where the position is accessed as:
+        # 'position_by_im_frame' is the average [x y theta] position by imaging frame
+        # 'velocity_by_im_frame' is the average [x y theta] velocity by imaging frame
+        position = self._mat_dict[position_by_im_frame_name]  # x,y,z
+
+        timestamps = self.get_timestamps()
+        assert (
+            len(timestamps) == position.shape[0]
+        ), f"The length of timestamps ({len(timestamps)}) must match the length of position ({position.shape[0]})."
+
+        reference_frame = "unknown"
+        if "SpatialSeries" in position_obj.spatial_series:
+            reference_frame = position_obj.spatial_series["SpatialSeries"].reference_frame
+        spatial_series_name = "SpatialSeriesByImFrame"
+        position_obj.create_spatial_series(
+            name=spatial_series_name,
+            data=H5DataIO(position[:, :-1], compression="gzip"),
+            description="The average x, y position by imaging frame.",
+            reference_frame=reference_frame,
```

Comment:
add conversion here as well 
```suggestion
            reference_frame=reference_frame,
            conversion=0.01,
```